packages/core/lib/core.ts

```diff
@@ -73,6 +73,11 @@ export function createClient<S extends CoreSchema, C extends Configuration, T> (
     start: (config: C | string) => {
       const configuration = validateConfig<S, C>(config, options.schema)
 
+      // sendPayloadChecksums is false by default unless custom endpoints are not specified
+      if (typeof config !== 'string' && !config.endpoint) {
+        configuration.sendPayloadChecksums = ('sendPayloadChecksums' in config && config.sendPayloadChecksums) || true
+      }
```

Comment:
Not keen on this but not sure of a better way. We need to work out what config options the user passed in to determine the behaviour. `const configuration = validateConfig<S, C>(config, options.schema)` does more than just validate, it creates a config with all values set based on the schema, using default values if not supplied.

Reply:
right now this is the place we're performing similar manipulation on the configuration, perhaps we could refactor the validation to include this, possibly separating out validation and 'cleaning' but I think for now this is fine